### PR TITLE
sdl2 2.24.0

### PR DESCRIPTION
**Pull request by FtZPetruska**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

New releases are now solely on the GitHub release page.

### CODE REVIEW

File: Formula/sdl2.rb [hunk 1]
```diff
@@ -41,7 +42,7 @@ def install
     # We have to do this because most build scripts assume that all SDL modules
     # are installed to the same prefix. Consequently SDL stuff cannot be
     # keg-only but I doubt that will be needed.
-    inreplace %w[sdl2.pc.in sdl2-config.in], "@prefix@", HOMEBREW_PREFIX
+    inreplace "sdl2.pc.in", "@prefix@", HOMEBREW_PREFIX
```
Comment (carlocab) [inline Formula/sdl2.rb:45]: I think we need to update this `inreplace` for `sdl2-config` too, since we don't want that to output a `Cellar` path.

Comment (FtZPetruska) [reply Formula/sdl2.rb:45]: Patching `sdl2-config` is no longer necessary, in this release prefix is computed at runtime depending on `sdl2-config`'s location. Testing locally, I got the following results:
```
% sdl2-config --cflags --libs --version
-I/opt/homebrew/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/lib -lSDL2
2.24.0
```

The `@prefix@` variable is no longer present in `sdl2-config.in` either, being replaced with this:
```sh
# Get the canonical path of the folder containing this script
bindir=$(cd -P -- "$(dirname -- "$0")" && printf '%s\n' "$(pwd -P)")

# Calculate the canonical path of the prefix, relative to the folder of this script
prefix=$(cd -P -- "$bindir/@bin_prefix_relpath@" && printf '%s\n' "$(pwd -P)")
```

Comment (carlocab) [reply Formula/sdl2.rb:45]: What happens if I call it as `/opt/homebrew/opt/sdl2/bin/sdl2-config --cflags --libs`?

Comment (FtZPetruska) [reply Formula/sdl2.rb:45]: In this case it does point to the Cellar version:
```
% /opt/homebrew/opt/sdl2/bin/sdl2-config --cflags --libs
-I/opt/homebrew/Cellar/sdl2/2.24.0/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/Cellar/sdl2/2.24.0/lib -lSDL2
```

Comment (carlocab) [reply Formula/sdl2.rb:45]: It sounds like we still need to patch it then. Maybe. What about
```
export PATH="/opt/homebrew/opt/sdl2/bin:$PATH"
sdl2-config --cflags --libs
```
?

File: Formula/sdl2.rb [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+  sha256 "91e4c34b1768f92d399b078e171448c6af18cafda743987ed2064a28954d6d97"
   license "Zlib"
 
   livecheck do
-    url "https://www.libsdl.org/download-2.0.php"
-    regex(/href=.*?SDL2[._-]v?(\d+(?:\.\d+)+)\.t/i)
+    url :stable
+    strategy :github_latest
+    regex(%r{href=.*?/tag/release[._-]v?(\d+(?:\.\d+)+)["' >]}i)
   end
```
Comment (carlocab) [inline Formula/sdl2.rb:12]:
```suggestion
  license "Zlib"
```
Just remove the `livecheck` block. The default works fine here.

File: Formula/sdl2.rb (hunk 1 above)
Comment (FtZPetruska) [reply Formula/sdl2.rb:45]: It still returns the cellar path unfortunately:
```
% export PATH="/opt/homebrew/opt/sdl2/bin:$PATH"
% sdl2-config --cflags --libs
-I/opt/homebrew/Cellar/sdl2/2.24.0/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/Cellar/sdl2/2.24.0/lib -lSDL2
```

Comment (carlocab) [reply Formula/sdl2.rb:45]: Hmmm, okay. Let's leave this be for now, but we might need to patch it later on.

Comment (FtZPetruska) [reply Formula/sdl2.rb:39]: The following patch restores the old behaviour:
```
diff --git a/Formula/sdl2.rb b/Formula/sdl2.rb
index fa8ff7a6050..d491355f38f 100644
--- a/Formula/sdl2.rb
+++ b/Formula/sdl2.rb
@@ -37,6 +37,7 @@ class Sdl2 < Formula
     # are installed to the same prefix. Consequently SDL stuff cannot be
     # keg-only but I doubt that will be needed.
     inreplace "sdl2.pc.in", "@prefix@", HOMEBREW_PREFIX
+    inreplace "sdl2-config.in", /bindir=.+\n/, "bindir=#{HOMEBREW_PREFIX}/bin"
 
     system "./autogen.sh" if build.head?
```

With it, `sdl2-config --cflags --libs` always returns:
```
-I/opt/homebrew/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/lib -lSDL2
```

Would you like the patch to be applied before merging ?

Comment (carlocab) [reply Formula/sdl2.rb:39]: Thanks for working that out. I'm inclined to leave it be for now to see what it breaks (if anything).

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**